courses/models.py

```diff
@@ -719,6 +731,13 @@ class BaseCertificate(models.Model):
 
     user = models.ForeignKey(User, null=False, on_delete=models.CASCADE)
     uuid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+    is_revoked = models.BooleanField(
+        default=False,
+        help_text="Indicates whether or not the certificate is revoked",
+        verbose_name="revoked",
+    )
+    objects = NonRevokedCertificates()
+    all_objects = models.Manager()
```

Comment:
I suggest moving these out of the abstract base class and adding it specifically to the concrete subclasses.